src/Task/ProductModel/BatchProductModelTask.php

```diff
@@ -52,64 +54,90 @@ public function __construct(
      */
     public function __invoke(PipelinePayloadInterface $payload): PipelinePayloadInterface
     {
+        if ($this->retryCount === $this->maxRetryCount) {
+            return $payload;
+        }
+
         $this->logger->debug(self::class);
         $this->type = $payload->getType();
         $this->logger->notice(Messages::createOrUpdate($this->type));
 
         $query = $this->getSelectStatement($payload);
-        /** @var Result $queryResult */
         $queryResult = $query->executeQuery();
 
         while ($results = $queryResult->fetchAll()) {
+            /** @var array $result */
             foreach ($results as $result) {
+                $isSuccess = false;
+
                 /** @var array $resource */
                 $resource = json_decode($result['values'], true);
 
-                $this->handleProductGroup($resource);
-
-                try {
-                    $this->dispatcher->dispatch(new BeforeProcessingProductEvent($resource));
-
-                    $this->entityManager->beginTransaction();
-
-                    if ($this->isProductProcessableChecker->check($resource)) {
-                        $product = $this->process($resource);
-                        $this->dispatcher->dispatch(new AfterProcessingProductEvent($resource, $product));
+                do {
+                    try {
+                        $this->handleProductModel($resource);
+                        $isSuccess = true;
+                    } catch (ORMInvalidArgumentException $ormInvalidArgumentException) {
+                        ++$this->retryCount;
+                        usleep($this->retryWaitTime);
+
+                        $this->logger->error('Retrying import', [
+                            'product' => $result,
+                            'retry_count' => $this->retryCount,
+                            'error' => $ormInvalidArgumentException->getMessage(),
+                        ]);
+
+                        $this->entityManager = $this->getNewEntityManager();
+                    } catch (\Throwable $throwable) {
+                        ++$this->retryCount;
+                        usleep($this->retryWaitTime);
+
+                        $this->logger->error('Error importing product', [
+                            'message' => $throwable->getMessage(),
+                            'trace' => $throwable->getTraceAsString(),
+                        ]);
+
+                        $this->entityManager = $this->getNewEntityManager();
                     }
+                } while (false === $isSuccess && $this->retryCount !== $this->maxRetryCount);
```

Comment:
I think we have a problem here. 
Because of `private int $retryCount = 0,` means that, from zero to n, you have n+1 loops.
The other problem is that if the last loop is successful (I have the impression) we enter an infinite loop. 
So I suggest, something like that : 

```suggestion
                } while (false === $isSuccess && $this->retryCount < $this->maxRetryCount);
```